DP/GuessNumberHigherOrLowerII.swift

```diff
@@ -0,0 +1,33 @@
+/**
+ * Question Link: https://leetcode.com/problems/guess-number-higher-or-lower-ii/
+ * Primary idea: Typical DP problem
+ * For each guess:
+ * 1) Get the worse case of every choice to guarantee the result
+ * 2) Get the least cost among the guaranteed results
+ * Time Complexity: O(nlogn), Space Complexity: O(n^2)
+ */
+
+class GuessNumberHigherOrLowerII {
+    func getMoneyAmount(_ n: Int) -> Int {
+        var table = [[Int]](repeatElement([Int](repeatElement(0, count: n + 1)), count: n + 1))
+        return DP(&table, 1, n)
+    }
+    
+    func DP(_ table: inout [[Int]], _ s: Int, _ e: Int) -> Int {
```

Comment:
please declare it as private or fileprivate, this should not be visible from outside